src/main/java/com/gitrekt/resort/model/entities/Booking.java

```diff
@@ -0,0 +1,94 @@
+package com.gitrekt.resort.model.entities;
+
+import java.util.Date;
+import java.util.List;
+import javax.persistence.CascadeType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+import org.hibernate.annotations.CreationTimestamp;
+
+public class Booking {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(cascade = CascadeType.ALL)
+    private Guest guest;
+
+    @CreationTimestamp
+    @Temporal(TemporalType.TIMESTAMP)
+    private Date checkInDate;
+
+    @CreationTimestamp
+    @Temporal(TemporalType.TIMESTAMP)
+    private Date checkOutDate;
+
+    @ManyToOne(cascade = CascadeType.ALL)
+    private Bill bill;
+
+    private String confirmationNumber;
+
+    private String specialInstructions;
+
+    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<Package> packages;
+
+    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<Room> bookedRooms;
+
+    /**
+     * DO NOT CALL THIS CONSTRUCTOR. IT IS INTENDED FOR USE BY HIBERNATE ONLY.
+     */
+    protected Booking() {
+        // REQUIRED BY HIBERNATE
+    }
+
+    public Booking(Guest guest, Date checkInDate, Date checkOutDate, Bill bill,
+            String confirmationNumber, String specialInstructions,
```

Comment:
We probably need to redesign this constructor, but for now it's okay. I've realized we have some issues in our design, and they might cause problems. Don't worry about it right now - we can fix it once we have decided how to.